### PR TITLE
refactor(keychain): add missing type definitions for keychain module

### PR DESCRIPTION
**Pull request by marcoskichel**
## Description

Add some missing type defs to the keychain module

### CODE REVIEW

File: features/keychain/api/index.d.ts [hunk 1]
```diff
@@ -4,6 +4,17 @@ import BN from 'bn.js'
 type SeedId = string
 type KeySource = { seedId: SeedId; keyId: KeyIdentifier }
 
+type Tx = Record<string, unknown>
```
Comment (marcoskichel) [inline features/keychain/api/index.d.ts:7]: Do we already have this defined somewhere?

Comment (mvayngrib) [reply features/keychain/api/index.d.ts:7]: https://github.com/ExodusMovement/exodus-hydra/blob/36b4edd68d5e99993194375f1c685fe6228b2b28/features/tx-signer/src/module/interfaces.ts#L31

Comment (marcoskichel) [reply features/keychain/api/index.d.ts]: Thanks I used the ones I found relevant in my recent PR

File: features/keychain/api/index.d.ts [hunk 2]
```diff
@@ -1,9 +1,20 @@
 import KeyIdentifier from '@exodus/key-identifier'
+import { UnsignedTransaction } from '@exodus/tx-signer'
```
Comment (kewde) [inline features/keychain/api/index.d.ts:2]:
```suggestion
import type { UnsignedTransaction } from '@exodus/tx-signer'
```

Comment (kewde) [reply features/keychain/api/index.d.ts:2]: same for other import

Comment (marcoskichel) [reply features/keychain/api/index.d.ts]: https://github.com/ExodusMovement/exodus-oss/pull/132/commits/901dea305df329c55708e3e9ff2aab8e0785c50e

**Comment by marcoskichel** [comment]
@mvayngrib should we merge this?

File: features/keychain/api/index.d.ts [hunk 3]
```diff
@@ -24,16 +35,67 @@ type PrivateKeys = {
...
+
 export interface KeychainApi {
+  syncTx(params: TxSignParams): Promise<Tx>
+  addSeed(seed: Buffer): Promise<SeedId>
+  removeAllSeeds(): void
+  clear(): void
+  clone(): KeychainApi
+  createSodiumEncryptor(keyId: string): Encryptor
+  createEd25519Signer(args: CreateSignerArgs): Ed25519Signer
+  createSecp256k1Signer(args: CreateSignerArgs): Secp256k1Signer
```
Comment (diegomura) [inline features/keychain/api/index.d.ts:91]: I see methods here that aren't part of the keychain api. Why is that?
[Here](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/api/index.js#L3) i see keychain only has:

- exportKey
- arePrivateKeysLocked
- sodium
- ed25519
- secp256k1

Comment (marcoskichel) [reply features/keychain/api/index.d.ts:91]: Oh, good catch, my bad, I was looking at the [keychain module](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/module/keychain.js#L25), but now I am confused with this [module/index.d.ts](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/module/index.d.ts#L1) file that we have, why is it exporting the actual modules and not type definitions, am I missing something?

Comment (mvayngrib) [reply features/keychain/api/index.d.ts:91]: cc @kewde

Comment (kewde) [reply features/keychain/api/index.d.ts:91]: I'm not sure actually, we can make it type definitions I suppose

File: features/keychain/api/index.d.ts
Comment (kewde) [inline features/keychain/api/index.d.ts:1]: Please export all the types that seems useful for consumers, you never know when you need em

File: features/keychain/api/index.d.ts (hunk 3 above)
Comment (sparten11740) [reply features/keychain/api/index.d.ts:91]: we should consider migrating keychain to typescript, it will take away the manual type maintenance burden.

**Pull request closed without merging**